### PR TITLE
fix: [CDS-41642]: Fix NullPointerException in `aggregateStatus` method of GraphRenderer class

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
Adds a check for execution status with `null` values, so not to throw NullPointerException in the `aggregateStatus` method of GraphRenderer class.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36386)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36386)

**Comment by vcalasansh** [comment]
trigger messagecheck

File: 400-rest/src/main/java/software/wings/service/impl/GraphRenderer.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ static ExecutionStatus aggregateStatus(List<ExecutionStatus> statuses) {
       }
     }
     final Optional<ExecutionStatus> notNewActiveStatuses =
-        activeStatuses.stream().filter(status -> status != NEW).findFirst();
+        activeStatuses.stream().filter(status -> status != null && status != NEW).findFirst();
```
Comment (lasdox) [inline 400-rest/src/main/java/software/wings/service/impl/GraphRenderer.java:125]:
```suggestion
        activeStatuses.stream().filter(status -> asList(NEW, null).contains(status)).findFirst();
```

Comment (vcalasansh) [reply 400-rest/src/main/java/software/wings/service/impl/GraphRenderer.java]: Made the change you suggested. Thank you!

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36386)* status: :shipit: complete! ✓

**Pull request merged**